action.yml

```diff
@@ -0,0 +1,118 @@
+name: 'Publish to NeoFS Github Action'
+description: 'This action allows you to publish files, tests reports, static web pages and etc. to NeoFS'
+author: 'Neo SPCC'
+branding:
+  icon: 'server'
+  color: 'green'
+inputs:
+  NEOFS_WALLET:
+    description: N3 wallet for NeoFS network access
+    required: true
+  NEOFS_WALLET_PASSWORD:
+    description: N3 wallet password
+    required: true
+  NEOFS_NETWORK_DOMAIN:
+    description: Rpc endpoint domain address
+    required: false
+    default: 'st1.storage.fs.neo.org'
+  NEOFS_HTTP_GATE:
+    description: HTTP Gateway domain address
+    required: false
+    default: 'http.fs.neo.org'
+  STORE_OBJECTS_CID:
+    description: Container ID for your data
+    required: true
+  PATH_TO_FILES_DIR:
+    description: Path to the directory with the files to be pushed
+    required: true
+  NEOFS_ATTRIBUTES:
+    description: User attributes in form of Key1=Value1,Key2=Value2
+    required: false
+  URL_PREFIX:
+    description: Prefix to the url address for each of the files(objects)
+    required: false
+  LIFETIME:
+    description: Number of epochs for object to stay valid
+    required: false
+    default: 0
+
+outputs:
+  OUTPUT_CONTAINER_URL:
+    description: Container URL
+    value: ${{ steps.put_files.outputs.container_url }}
+
+runs:
+  using: composite
+  steps:
+    - name: Download latest stable neofs-cli for uploading reports to NeoFS
+      uses: dsaltares/fetch-gh-release-asset@1.1.1
+      with:
+        repo: 'nspcc-dev/neofs-node'
+        version: 'tags/v0.38.1'
+        file: 'neofs-cli-amd64'
+        target: 'neofs/neofs-cli'
+
+    - name: Chmod latest stable neofs-cli
+      shell: bash
+      if: runner.os == 'Linux'
+      run: |
+        sudo chmod a+x neofs-cli
+      working-directory: neofs
+
+    - name: Enable stable neofs-cli
+      shell: bash
+      if: always()
+      run: |
+        echo "$(pwd)" >> $GITHUB_PATH
+      working-directory: neofs
+
+    - name: Create wallet
+      shell: bash
+      env:
+        NEOFS_WALLET: ${{ inputs.NEOFS_WALLET }}
+      run: |
+        echo "$NEOFS_WALLET" | base64 -d > wallet.json
+
+    - name: Define expiration
+      shell: bash
+      env:
+        NEOFS_NETWORK_DOMAIN: ${{ inputs.NEOFS_NETWORK_DOMAIN }}
+        LIFETIME: ${{ inputs.LIFETIME }}
+      run: |
+        if [[ "$LIFETIME" =~ ^[0-9]+$ ]]; then           
+            if [[ "$LIFETIME" -eq "0" ]]; then
+              EXP_EPOCH=0
+            else
+              CURRENT_EPOCH=$(neofs-cli netmap epoch --rpc-endpoint $NEOFS_NETWORK_DOMAIN:8080)
+              EXP_EPOCH=$((LIFETIME + CURRENT_EPOCH))
+            fi
+          else
+            echo "Error: LIFETIME $LIFETIME is not a valid number."
+            exit 1
+          fi
```

Comment:
> Are you sure that this shouldn't be a part of the uploading python script? It can be both, but I think that this code will bring more benefit if moving to the uploading script.

Fixed

Reply:
> Are you sure that this shouldn't be a part of the uploading python script? It can be both, but I think that this code will bring more benefit if moving to the uploading script.

Fixed